### ISSUE TITLE
receive error during "pub update" command

### ISSUE THREAD

**Issue by Abaranenko** [dart-lang/pub#2224]
C:\Work\dart\grpc-dart\example\metadata>pub --trace get
Resolving dependencies...
The method 'formatDescription' was called on null.
Receiver: null
Tried calling: formatDescription("metadata")
dart:core                                   Object.noSuchMethod
package:pub/src/solver/report.dart 241:35   SolveReport._writeId
package:pub/src/solver/report.dart 190:5    SolveReport._reportPackage
package:pub/src/solver/report.dart 110:9    SolveReport._reportChanges
package:pub/src/solver/report.dart 44:5     SolveReport.show
package:pub/src/solver/result.dart 91:65    SolveResult.showReport
package:pub/src/entrypoint.dart 224:12      Entrypoint.acquireDependencies
===== asynchronous gap ===========================
dart:async                                  Future.catchError
package:pub/src/utils.dart 109:52           captureErrors.<fn>
package:stack_trace                         Chain.capture
package:pub/src/utils.dart 122:11           captureErrors
package:pub/src/command_runner.dart 171:13  PubCommandRunner.runCommand
dart:async                                  _completeOnAsyncReturn
package:pub/src/command_runner.dart         PubCommandRunner._validatePlatform
dart:async                                  _completeOnAsyncReturn
package:pub/src/io.dart                     runProcess.<fn>
---- Log transcript ----
FINE: Pub 2.5.1
IO  : Spawning "cmd /c ver" in C:\Work\dart\grpc-dart\example\metadata\.
IO  : Finished ver. Exit code 0.
    | stdout:
    | |
    | | Microsoft Windows [Version 10.0.18362.356]
    | Nothing output on stderr.
MSG : Resolving dependencies...
SLVR: fact: metadata is 0.0.0
SLVR: derived: metadata
SLVR: fact: metadata depends on protobuf ^1.0.0
SLVR: fact: metadata depends on async ^2.2.0
SLVR: fact: metadata depends on grpc from path
SLVR:   selecting metadata
SLVR:   derived: grpc from path
SLVR:   derived: async ^2.2.0
SLVR:   derived: protobuf ^1.0.0
SLVR:   fact: grpc 2.1.2 from path depends on http ^0.12.0
SLVR:   fact: grpc 2.1.2 from path depends on googleapis_auth ^0.2.7
SLVR:   fact: grpc 2.1.2 from path depends on async ^2.2.0
SLVR:   fact: grpc 2.1.2 from path depends on http2 ^1.0.0
SLVR:   fact: grpc 2.1.2 from path depends on meta ^1.1.6
SLVR:     selecting grpc 2.1.2 from path
SLVR:     derived: meta ^1.1.6
SLVR:     derived: http2 ^1.0.0
SLVR:     derived: googleapis_auth ^0.2.7
SLVR:     derived: http ^0.12.0
SLVR:     fact: async 2.3.0 depends on collection ^1.5.0
SLVR:       selecting async 2.3.0
SLVR:       derived: collection ^1.5.0
SLVR:       fact: protobuf 1.0.0 depends on fixnum ^0.10.9
SLVR:         selecting protobuf 1.0.0
SLVR:         derived: fixnum ^0.10.9
SLVR:           selecting meta 1.1.7
SLVR:             selecting http2 1.0.0
SLVR:             fact: googleapis_auth 0.2.10 depends on crypto >=0.9.2 <3.0.0
SLVR:             fact: googleapis_auth 0.2.10 depends on http >=0.11.3+17 <0.13.0
SLVR:               selecting googleapis_auth 0.2.10
SLVR:               derived: crypto >=0.9.2 <3.0.0
SLVR:               fact: http 0.12.0+2 depends on http_parser >=0.0.1 <4.0.0
SLVR:               fact: http 0.12.0+2 depends on pedantic ^1.0.0
SLVR:               fact: http 0.12.0+2 depends on async >=1.10.0 <3.0.0
SLVR:               fact: http 0.12.0+2 depends on path >=0.9.0 <2.0.0
SLVR:                 selecting http 0.12.0+2
SLVR:                 derived: path >=0.9.0 <2.0.0
SLVR:                 derived: pedantic ^1.0.0
SLVR:                 derived: http_parser >=0.0.1 <4.0.0
SLVR:                   selecting collection 1.14.12
SLVR:                     selecting fixnum 0.10.9
SLVR:                     fact: crypto 2.1.3 depends on convert >=1.0.0 <3.0.0
SLVR:                     fact: crypto 2.1.3 depends on collection ^1.0.0
SLVR:                     fact: crypto 2.1.3 depends on typed_data ^1.0.0
SLVR:                       selecting crypto 2.1.3
SLVR:                       derived: typed_data ^1.0.0
SLVR:                       derived: convert >=1.0.0 <3.0.0
SLVR:                         selecting path 1.6.4
SLVR:                           selecting pedantic 1.8.0+1
SLVR:                           fact: http_parser 3.1.3 depends on charcode ^1.1.0
SLVR:                           fact: http_parser 3.1.3 depends on source_span ^1.0.0
SLVR:                           fact: http_parser 3.1.3 depends on string_scanner >=0.0.0 <2.0.0
SLVR:                           fact: http_parser 3.1.3 depends on collection >=0.9.1 <2.0.0
SLVR:                           fact: http_parser 3.1.3 depends on typed_data ^1.1.0
SLVR:                             selecting http_parser 3.1.3
SLVR:                             derived: typed_data ^1.1.0
SLVR:                             derived: string_scanner >=0.0.0 <2.0.0
SLVR:                             derived: source_span ^1.0.0
SLVR:                             derived: charcode ^1.1.0
SLVR:                               selecting typed_data 1.1.6
SLVR:                               fact: convert 2.1.1 depends on charcode ^1.1.0
SLVR:                               fact: convert 2.1.1 depends on typed_data ^1.1.0
SLVR:                                 selecting convert 2.1.1
SLVR:                                 fact: string_scanner 1.0.5 depends on charcode ^1.1.0
SLVR:                                 fact: string_scanner 1.0.5 depends on source_span ^1.4.0
SLVR:                                 fact: string_scanner 1.0.5 depends on meta ^1.1.0
SLVR:                                   selecting string_scanner 1.0.5
SLVR:                                   derived: source_span ^1.4.0
SLVR:                                   fact: source_span 1.5.5 depends on charcode ^1.0.0
SLVR:                                   fact: source_span 1.5.5 depends on term_glyph ^1.0.0
SLVR:                                   fact: source_span 1.5.5 depends on path ^1.2.0
SLVR:                                     selecting source_span 1.5.5
SLVR:                                     derived: term_glyph ^1.0.0
SLVR:                                       selecting charcode 1.1.2
SLVR:                                         selecting term_glyph 1.1.0
SLVR: Version solving took 0:00:00.135365 seconds.
    | Tried 1 solutions.
FINE: Resolving dependencies finished (0.1s).
ERR : The method 'formatDescription' was called on null.
    | Receiver: null
    | Tried calling: formatDescription("metadata")
FINE: Exception type: NoSuchMethodError
ERR : dart:core                                   Object.noSuchMethod
    | package:pub/src/solver/report.dart 241:35   SolveReport._writeId
    | package:pub/src/solver/report.dart 190:5    SolveReport._reportPackage
    | package:pub/src/solver/report.dart 110:9    SolveReport._reportChanges
    | package:pub/src/solver/report.dart 44:5     SolveReport.show
    | package:pub/src/solver/result.dart 91:65    SolveResult.showReport
    | package:pub/src/entrypoint.dart 224:12      Entrypoint.acquireDependencies
    | ===== asynchronous gap ===========================
    | dart:async                                  Future.catchError
    | package:pub/src/utils.dart 109:52           captureErrors.<fn>
    | package:stack_trace                         Chain.capture
    | package:pub/src/utils.dart 122:11           captureErrors
    | package:pub/src/command_runner.dart 171:13  PubCommandRunner.runCommand
    | dart:async                                  _completeOnAsyncReturn
    | package:pub/src/command_runner.dart         PubCommandRunner._validatePlatform
    | dart:async                                  _completeOnAsyncReturn
    | package:pub/src/io.dart                     runProcess.<fn>
---- End log transcript ----

**Comment by sigurdm** [comment]
I believe this error can no longer happen (we are now null-safe).

Please open a new issue if this can still be reproduced with a recent sdk.

Also please include enough details for us to reproduce.
